src/lib/sha256.ts

```diff
@@ -0,0 +1,41 @@
+/**
+ * Code originally from the example at
+ * https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/digest
+ *
+ * Note: Using SHA256 instead of the previously used MD5 due to
+ * the following comment found at the documentation page linked above:
+ *
+ * Warning: Older insecure hash functions, like MD5, are not supported
+ * by this method. Even a supported method, SHA-1, is considered weak,
+ * has been broken and should be avoided for cryptographic applications.
```

Comment:
This doesn't really matter for us since we aren't using the hash in a cryptographic application. We only care about performance and probability of collisions, and both MD5 and SHA256 are sufficient for the number of files we expect to hash. The only reason to stay with MD5 is consistency with the older datasets, as it would allow us to easily de-duplicate data across new and old crawls.

However, it appears that Firefox doesn't expose a built-in way to compute MD5 hashes in web / webextension JS. Let's keep sha256, as that's preferable to adding an external library for MD5. Anyone who wants to de-duplicate JS files across old and new crawls can pay the one-time cost of converting the old databases to sha256.

Reply:
Well put :)